service/src/query_processor.rs

```diff
@@ -100,6 +105,7 @@ pub struct Response<T> {
 #[derive(Debug)]
 pub struct FreeQuery {
     pub subgraph_deployment_id: SubgraphDeploymentID,
+    // TODO: Use Arc<String> instead of String?
```

Comment:
maybe worthwhile, especially if you think the query string is shared across threads or parts of the program, or if it is used multiple times, or should be long living

Reply:
Yup, in particular I noticed that
https://github.com/hopeyen/indexer-service-rs/blob/c604c7a258fad37ef74abd20823571d5f4db9e3e/service/src/graph_node.rs#L28-L32
needs the cloning.

Reply:
I played around, went down the rabbit hole, and gave up: [8b9b531](https://github.com/graphops/indexer-service-rs/pull/21/commits/8b9b531ea6b59ef34563dc18ca4f9590727890a1)

Reply:
haha we could leave the TODO comment in for a future refactor potentially?

Reply:
Well, I'm not even sure that Arc is the solution here. So I'm of the opinion of sticking to the YAGNI rule, we can do something about it when we'll actually need it.